### PR TITLE
Add factory for router

### PR DESCRIPTION
**Pull request by matthew-white**

### CODE REVIEW

**Review by matthew-white — COMMENTED** [review]
Notes from interactive code review with @ktuite

File: test/util/lifecycle.js [hunk 1]
```diff
@@ -109,3 +97,35 @@ export const mount = (component, options = {}) => {
...
+export const mergeMountOptions = (options, defaults) => {
+  if (options == null) return defaults;
+  const merged = { ...defaults, ...options };
+  for (const path of optionsToMerge) {
+    const lens = lensPath(path);
+    const option = view(lens, options);
+    if (option != null) {
+      const defaultOption = view(lens, defaults);
+      if (defaultOption != null) {
+        let obj = merged;
```
Comment (matthew-white) [inline test/util/lifecycle.js:123]: Rename `obj` to `parent`.

**Review by matthew-white — COMMENTED** [review]
Notes from interactive review with @ktuite

**Review by ktuite — APPROVED** [review]

**Comment by matthew-white** [comment]
I'm going to go ahead and merge #579 into this PR so that changes I make here after review don't result in conflicts.

**Pull request merged**